src/pb_tensor.cc

```diff
@@ -305,11 +339,30 @@ PbTensor::Memory()
 
 #ifdef TRITON_PB_STUB
 std::shared_ptr<PbTensor>
-PbTensor::FromDLPack(const std::string& name, const py::capsule& dlpack_tensor)
+PbTensor::FromDLPack(const std::string& name, const py::object& tensor)
 {
   if (name == "") {
     throw PythonBackendException("Tensor name cannot be an empty string.");
   }
+  if(py::isinstance<py::capsule>(tensor)){
+    return FromDLPackCapsule(name, tensor);
+  } else if (py::hasattr(tensor, "__dlpack__")) {
+    // Python backend does not support async executions on PbTensor objects,
+    // thus we can count on the default `stream=None` argument.
+    // For CPU, `stream=None` is the only accepted argument 
+    // according to array API. For GPU, when `stream=None`  producer must 
+    // assume the legacy default stream.
+    // Reference:
+    // https://data-apis.org/array-api/latest/API_specification/generated/array_api.array.__dlpack__.html
+    return FromDLPackCapsule(name, tensor.attr("__dlpack__")());
```

Comment:
I think based on the API descriptions we should provide -1?

>  If stream is -1, the value may be used by the consumer to signal “producer must not perform any synchronization”.

Reply:
We can do this. The way it is now , we are telling to the producer that he can assume default stream, which we are on now. I believe this is also an indication, that sync is not required, since python backend does not allow async execution.

Reply:
I didn't infer that from the API description. It just mentions that the tensor would be on the default stream but the consumer would still need to synchronize even if it is on the default stream.

Reply:
I see, I believe PyTorch treats the passed `None` as no sync as well, I probably was influenced by their implementation a little bit. I'll make a change to pass `-1` for no sync.

Reply:
I'm passing `stream=None` in Revision 1, this way if tensor is on CPU, this is the only option that allowed, and if a tensor is on GPU, legacy stream is assumed according to API standard